apps/teamsjs-cc-app/tabs/src/components/Tab.tsx

```diff
@@ -0,0 +1,330 @@
+          { key: '12-1', content: <><Fluent.FilesTxtIcon />Pages.CurrentApp</>, value: 'Pages.CurrentApp' },
+          { key: '12-2', content: IsPagesCurrentAppSupported() },
+          { key: '12-3', content: <PagesCurrent />, className: 'ui_action' }
+        ],
+      },
+      {
+        key: 13,
+        items: [
+          { key: '13-1', content: <><Fluent.FilesTxtIcon />Pages</>, value: 'Pages' },
+          { key: '13-2', content: PagesIsSupported() },
+          { key: '13-3', content: <Pages />, className: 'ui_action' }
+        ],
+      },
+      {
+        key: 14,
+        items: [
+          { key: '14-1', content: <><Fluent.AttendeeIcon />People</>, value: 'People' },
+          { key: '14-2', content: PeopleIsSupported() },
+          { key: '14-3', content: <People />, className: 'ui_action' }
+        ]
+      },
+      {
+        key: 15,
+        items: [
+          { key: '15-1', content: <><Fluent.ContactCardIcon />Profile</>, value: 'Profile' },
+          { key: '15-2', content: ProfileIsSupported() },
+          { key: '15-3', content: <Profile />, className: 'ui_action' }
+        ],
+      },
+      {
+        key: 16,
+        items: [
+          { key: '16-1', content: <><Fluent.SearchIcon />Search</>, value: 'Search' },
+          { key: '16-2', content: SearchIsSupported() },
+          { key: '16-3', content: <Search />, className: 'ui_action' }
+        ],
+      },
+      {
+        key: 17,
+        items: [
+          { key: '17-1', content: <><Fluent.ScreenshareIcon />Sharing</>, value: 'Sharing' },
+          { key: '17-2', content: SharingIsSupported() },
+          { key: '17-3', content: <Sharing />, className: 'ui_action' }
+        ],
+      },
+      {
+        key: 18,
+        items: [
+          { key: '18-1', content: <><Fluent.PanoramaIcon />Stage View</>, value: 'Stage View' },
+          { key: '18-2', content: StageViewIsSupported() },
+          { key: '18-3', content: <StageView />, className: 'ui_action' }
+        ],
+      },
+      {
+        key: 19,
+        items: [
+          { key: '19-1', content: <><Fluent.TeamsMonochromeIcon />Teams Core</>, value: 'Teams Core' },
+          { key: '19-2', content: TeamsCoreIsSupported() },
+          { key: '19-3', content: <TeamsCore />, className: 'ui_action' }
+        ],
+      },
+      {
+        key: 20,
+        items: [
+          { key: '20-1', content: <><Fluent.CallVideoIcon />Video</>, value: 'Video' },
+          { key: '20-2', content: VideoIsSupported() },
+          { key: '20-3', content: <Video />, className: 'ui_action' }
+        ],
+      },
+      {
+        key: 21,
+        items: [
+          { key: '21-1', content: <><Fluent.BriefcaseIcon />Web Storage</>, value: 'Web Storage' },
+          { key: '21-2', content: WebStorageIsSupported() },
+          { key: '21-3', content: <WebStorage />, className: 'ui_action' }
+        ],
+      }
+    ];
+  }
+
+  const updateCapabilityOnchange = (text: string) => {
```

Comment:
What is this function being used for? It looks like it's being used to regenerate the rows in SetData, turn off the ability to only see supported rows, and then returning all rows that contain text as a substring. Is this used to filter a list that shows up in UI somewhere based on user input?

Reply:
This is a search function, I will update the name to seachCapability().